lib_eio_linux/eio_linux.ml

```diff
@@ -1160,11 +1160,87 @@ let net = object
   method getnameinfo = Eio_unix.getnameinfo
 end
 
+let pid_to_process close pid = object
+  inherit Eio.Process.process
+  method pid = pid
+  method status = 
+    match Eio_unix.run_in_systhread @@ fun () -> let v = Unix.waitpid [] pid in close (); v with
+    | _, WEXITED i -> Exited i
+    | _, WSIGNALED i -> Signaled i
+    | _, WSTOPPED i -> Stopped i
+  
+  method stop = 
+    Unix.kill pid Sys.sigkill
+end
+
+let source_or_std fd = function
+  | Some flow -> flow
+  | None -> (source (FD.of_unix_no_hook ~seekable:(FD.is_seekable fd) ~close_unix:true fd) :> Eio.Flow.source)
+
+let sink_or_std fd = function
+  | Some flow -> flow
+  | None -> (sink (FD.of_unix_no_hook ~seekable:(FD.is_seekable fd) ~close_unix:true fd) :> Eio.Flow.sink)
+
+let get_fd_or_err flow =
+  match get_fd_opt flow with
+  | Some fd -> 
+    Unix.dup ~cloexec:false (FD.to_unix `Peek fd)
+  | None -> failwith "Currently only flows backed by file descriptors are supported!"
+
+let open_null flags = lazy (Unix.openfile "/dev/null" flags 0o666)
+
+let dev_null_in = open_null [ Unix.O_RDONLY; Unix.O_CLOEXEC ]
+
+let dev_null_out = open_null [ Unix.O_WRONLY; Unix.O_CLOEXEC ]
+
+let process = object
+  inherit Eio.Process.t
+
+  (* TODO: Is Sys.chdir cwd domain-safe, ? *)
+  method spawn ~sw ?cwd ?stderr ?stdout ?stdin cmd args =
+    let stdin = source_or_std (Lazy.force dev_null_in) stdin |> get_fd_or_err in
+    let stdout = sink_or_std (Lazy.force dev_null_out) stdout |> get_fd_or_err in
+    let stderr = sink_or_std (Lazy.force dev_null_out) stderr |> get_fd_or_err in
+    let pid =
+      Option.iter Sys.chdir cwd;
+      Unix.create_process cmd
+        (Array.of_list args) 
+        stdin stdout stderr
+    in
+    let close () =
+      Unix.close stdin;
+      Unix.close stdout;
+      Unix.close stderr
+    in
+    let process = pid_to_process close pid in
+    let cleanup () =
```

Comment:
Should kill it here I think. We know that no fiber is waiting for the result, or the switch wouldn't be getting released.

Reply:
Hopefully done in https://github.com/ocaml-multicore/eio/pull/330/commits/0127acc27b8db6dbb4ae3b29ae3546597f4e98d1 -- it tries to stop the process and catches `ESRCH` if it is already gone, would it be better to remove the hook if `status` is called instead?

Reply:
Cleaning up now stops the process 👍

Reply:
Not sure about whether we want an exception or not. I'm tempted to put the `waitpid` in its own fiber inside the switch, in which case it would stop the switch from exiting (unless there was an error). Then forking a process behaves like forking a fiber, keeping the switch alive.